src/st/st-background-effect.c

```diff
@@ -55,377 +77,283 @@ static const gchar *box_blur_glsl_shader =
-
-                  bump_step[0] = 1.0f / (self->bumptex_width_i);
-                  bump_step[1] = 1.0f / (self->bumptex_height_i);
-
-                  cogl_pipeline_set_uniform_float (self->pipeline0,
-                                                   self->bump_step_uniform,
-                                                   2,
-                                                   1,
-                                                   bump_step);
-                }
-
-              if (self->bg_sub_texture != NULL)
-                {
-                  cogl_handle_unref (self->bg_sub_texture);
-                  self->bg_sub_texture = NULL;
-                }
-
-              self->bg_sub_texture = st_cogl_texture_new_with_size_wrapper (self->bg_width_i, self->bg_height_i,
-                                                                            COGL_TEXTURE_NO_SLICING,
-                                                                            COGL_PIXEL_FORMAT_RGBA_8888_PRE);
-
-              cogl_pipeline_set_layer_texture (self->pipeline0, 0, self->bg_texture);
-
-              if (self->pixel_step_uniform1 > -1)
-                {
-                  gfloat pixel_step[3];
-
-                  pixel_step[0] = 1.0f / (self->bg_width_i);
-                  pixel_step[1] = 1.0f / (self->bg_height_i);
-                  pixel_step[2] = 1.0f;
-
-                  cogl_pipeline_set_uniform_float (self->pipeline1,
-                                                   self->pixel_step_uniform1,
-                                                   3,
-                                                   1,
-                                                   pixel_step);
-                }
-
-              if (self->pixel_step_uniform2 > -1)
-                {
-                  gfloat pixel_step[3];
-
-                  pixel_step[0] = 1.0f / (self->fg_width_i);
-                  pixel_step[1] = 1.0f / (self->fg_height_i);
-                  pixel_step[2] = 2.0f;
-
-                  cogl_pipeline_set_uniform_float (self->pipeline3,
-                                                   self->pixel_step_uniform2,
-                                                   3,
-                                                   1,
-                                                   pixel_step);
-                }
+      cogl_pipeline_set_layer_texture (self->pipeline1, 0, self->bg_texture);
 
-            }
+      cogl_framebuffer_draw_rectangle (fb, self->pipeline1, 0, 0, self->bg_width,self->bg_height);
+    }
 
-          cogl_pipeline_set_layer_texture (self->pipeline2, 0, fg_texture);
-          cogl_pipeline_set_layer_texture (self->pipeline3, 0, fg_texture);
-          cogl_pipeline_set_layer_texture (self->pipeline4, 0, fg_texture);
+    return TRUE;
+}
 
-        }
-      return TRUE;
-    }
-  else
-    {
-      return FALSE;
-    }
+static gboolean
+st_background_blur_effect_pre_paint (ClutterEffect *effect)
+{
+  return TRUE;
```

Comment:
What is the purpose of this function?

Reply:
Doesn't appear to be necessary, removed.